benchmarks/bench_hist_gradient_boosting_threading.py

```diff
@@ -0,0 +1,316 @@
+from time import time
+import argparse
+import os
+from pprint import pprint
+
+import numpy as np
+from threadpoolctl import threadpool_limits
+import sklearn
+from sklearn.model_selection import train_test_split
+# To use this experimental feature, we need to explicitly ask for it:
+from sklearn.experimental import enable_hist_gradient_boosting  # noqa
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.ensemble import HistGradientBoostingClassifier
+from sklearn.datasets import make_classification
+from sklearn.datasets import make_regression
+from sklearn.ensemble._hist_gradient_boosting.utils import (
+    get_equivalent_estimator)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--n-leaf-nodes', type=int, default=31)
+parser.add_argument('--n-trees', type=int, default=10)
+parser.add_argument('--lightgbm', action="store_true", default=False,
+                    help='also benchmark lightgbm')
+parser.add_argument('--xgboost', action="store_true", default=False,
+                    help='also benchmark xgboost')
+parser.add_argument('--catboost', action="store_true", default=False,
+                    help='also benchmark catboost')
+parser.add_argument('--learning-rate', type=float, default=.1)
+parser.add_argument('--problem', type=str, default='classification',
+                    choices=['classification', 'regression'])
+parser.add_argument('--loss', type=str, default='default')
+parser.add_argument('--missing-fraction', type=float, default=0)
+parser.add_argument('--n-classes', type=int, default=2)
+parser.add_argument('--n-samples', type=int, default=int(1e6))
+parser.add_argument('--n-features', type=int, default=100)
+parser.add_argument('--max-bins', type=int, default=255)
+
+parser.add_argument('--print-params', action="store_true", default=False)
+parser.add_argument('--random-sample-weights', action="store_true",
+                    default=False,
+                    help="generate and use random sample weights")
+parser.add_argument('--plot', action="store_true", default=False,
+                    help='show a plot results')
+parser.add_argument('--plot-filename', default=None,
+                    help='filename to save the figure to disk')
+args = parser.parse_args()
+
+n_samples = args.n_samples
+n_leaf_nodes = args.n_leaf_nodes
+n_trees = args.n_trees
+lr = args.learning_rate
+max_bins = args.max_bins
+
+
+print("Data size: %d samples train, %d samples test."
+      % (n_samples, n_samples))
+print(f"n_features: {args.n_features}")
+
+
+def get_estimator_and_data():
+    if args.problem == 'classification':
+        X, y = make_classification(args.n_samples * 2,
```

Comment:
Should this just be `args.n_samples`?

(I don't even remember why I did that on the older benchmark)

Reply:
train / test split?